contracts/PBAB+Collabs/GenArt721CoreV2_PBAB_FLEX.sol

```diff
@@ -0,0 +1,893 @@
+        ] = _script;
+        projects[_projectId].scriptCount = projects[_projectId].scriptCount + 1;
+    }
+
+    /**
+     * @notice Updates script for project `_projectId` at script ID `_scriptId`.
+     * @param _projectId Project to be updated.
+     * @param _scriptId Script ID to be updated.
+     * @param _script Script to be added.
+     */
+    function updateProjectScript(
+        uint256 _projectId,
+        uint256 _scriptId,
+        string memory _script
+    ) public onlyUnlocked(_projectId) onlyArtistOrWhitelisted(_projectId) {
+        require(
+            _scriptId < projects[_projectId].scriptCount,
+            "scriptId out of range"
+        );
+        projects[_projectId].scripts[_scriptId] = _script;
+    }
+
+    /**
+     * @notice Updates external asset dependency for project `_projectId`.
+     * @param _projectId Project to be updated.
+     * @param _index Asset index.
+     * @param _cid Asset cid (Content identifier).
+     * @param _dependencyType Asset dependency type.
+     *  0 - IPFS
+     *  1 - ARWEAVE
+     */
+    function updateProjectExternalAssetDependency(
+        uint256 _projectId,
+        uint256 _index,
+        string calldata _cid,
+        ExternalAssetDependencyType _dependencyType
+    )
+        external
+        onlyUnlockedProjectExternalAssetDependencies(_projectId)
+        onlyArtistOrWhitelisted(_projectId)
+    {
+        require(
+            _index < projects[_projectId].externalAssetDependencyCount,
+            "Asset index out of range"
+        );
+        projects[_projectId].externalAssetDependencies[_index].cid = _cid;
+        projects[_projectId]
+            .externalAssetDependencies[_index]
+            .dependencyType = _dependencyType;
+        emit ExternalAssetDependencyUpdated(
+            _projectId,
+            _index,
+            _cid,
+            _dependencyType
+        );
+    }
+
+    /**
+     * @notice Removes external asset dependency for project `_projectId` at index `_index`.
+     * Assets with indices higher than `_index` can have their indices adjusted as a result of this operation.
+     * @param _projectId Project to be updated.
+     * @param _index Asset index
+     */
+    function removeProjectExternalAssetDependency(
+        uint256 _projectId,
+        uint256 _index
+    )
+        external
+        onlyUnlockedProjectExternalAssetDependencies(_projectId)
+        onlyArtistOrWhitelisted(_projectId)
+    {
+        require(
+            _index < projects[_projectId].externalAssetDependencyCount,
+            "Asset index out of range"
+        );
+
+        uint256 lastElementIndex = projects[_projectId]
+            .externalAssetDependencyCount - 1;
+
+        projects[_projectId].externalAssetDependencies[_index] = projects[
+            _projectId
+        ].externalAssetDependencies[lastElementIndex];
+        delete projects[_projectId].externalAssetDependencies[lastElementIndex];
```

Comment:
This is good and will reset the externalAssetDependencies.
A super insane quirk when deleting structs (that doesn't apply here, but thought you might find interesting):
https://docs.soliditylang.org/en/develop/types.html#data-location-and-assignment-behaviour

Specifically:
"delete has no effect on mappings (as the keys of mappings may be arbitrary and are generally unknown). So if you delete a struct, it will reset all members that are not mappings and also recurse into the members unless they are mappings. However, individual keys and what they map to can be deleted: If a is a mapping, then delete a[x] will delete the value stored at x."

😅 😅

Reply:
ahhh yes, saw that when I was transitioning. definitely funky heh